cli/src/test/java/com/liferay/blade/cli/command/UpdateCommandTest.java

```diff
@@ -0,0 +1,160 @@
+
+		Assert.assertFalse(
+			"currentVersion = " + currentVersion + " should NOT be updated to " + updatedVersion,
+			UpdateCommand.shouldUpdate(currentVersion, updatedVersion));
+	}
+
+	@Test
+	public void testCurrentMinorLessThanUpdatedMinor() {
+		String currentVersion = "12.1.9.SCHWIBBY";
+		String updatedVersion = "12.2.1-snapshot";
+
+		Assert.assertTrue(
+			"currentVersion = " + currentVersion + " should be updated to " + updatedVersion,
+			UpdateCommand.shouldUpdate(currentVersion, updatedVersion));
+	}
+
+	@Test
+	public void testCurrentMinorMoreThanUpdatedMinor() {
+		String currentVersion = "3.6.0.001810231234";
+		String updatedVersion = "3.5.9-SCHNAPS";
+
+		Assert.assertFalse(
+			"currentVersion = " + currentVersion + " should NOT be updated to " + updatedVersion,
+			UpdateCommand.shouldUpdate(currentVersion, updatedVersion));
+	}
+
+	@Test
+	public void testCurrentPatchLessThanUpdatedPatch() {
+		String currentVersion = "123.10.10.SCHOOBY";
+		String updatedVersion = "123.10.20-whiff";
+
+		Assert.assertTrue(
+			"currentVersion = " + currentVersion + " should be updated to " + updatedVersion,
+			UpdateCommand.shouldUpdate(currentVersion, updatedVersion));
+	}
+
+	@Test
+	public void testCurrentPatchMoreThanUpdatedPatch() {
+		String currentVersion = "3.5.9.001810231234";
+		String updatedVersion = "3.5.8.999999";
+
+		Assert.assertFalse(
+			"currentVersion = " + currentVersion + " should NOT be updated to " + updatedVersion,
+			UpdateCommand.shouldUpdate(currentVersion, updatedVersion));
+	}
+
+	@Test
+	public void testCurrentVersionWithNoManifest() throws Exception {
+		BladeTestResults bladeTestResults = TestUtil.runBlade(false, "version");
+
+		String errors = bladeTestResults.getErrors();
+
+		String expectedErrorMessage = "Could not determine version.";
+
+		Assert.assertEquals(expectedErrorMessage, errors.trim());
+	}
+
+	@Ignore
+	@Test
+	public void testGetUpdateVersionReleases() throws IOException {
+		String updateVersion = UpdateCommand.getUpdateVersion(false);
+
+		Assert.assertNotNull(updateVersion);
+
+		Assert.assertFalse("updateVersion does not look right.", updateVersion.isEmpty());
+	}
+
+	@Ignore
+	@Test
+	public void testTargetReleases() throws IOException {
+		boolean ok;
+
+		// assuming target is in releases and available
+
+		String url = UpdateCommand.getUpdateJarUrl(false);
+
+		// expect: valid update url from the releases repo
+
+		if (url.length() > 1) {
+			ok = true;
+		}
+		else {
+			ok = false;
+		}
+
+		Assert.assertTrue("url = " + url + " ... this does not look right.", ok);
+		Assert.assertTrue(
+			"url is not from releases repo.  url = " + url, url.contains(UpdateCommand.RELEASES_REPO_URL));
+	}
+
+	@Test
+	public void testTargetSnapshots() throws IOException {
+		boolean ok;
+
+		// assuming target is in snapshots and available
+
+		String url = UpdateCommand.getUpdateJarUrl(true);
+
+		// expect: valid update url from the snapshots repo
+
+		if (url.length() > 1) {
```

Comment:
What about `boolean ok = url.length() > 1;`

Reply:
Fixed for you.

Reply:
Oh, formatSource will not accept this.  It changes it back to the way I had it :-(